code/game/machinery/vending/_vending.dm

```diff
@@ -4,8 +4,8 @@
 /obj/machinery/vending
 	name = "Vendomat"
 	desc = "A generic vending machine."
-	icon = 'icons/obj/vending.dmi'
-	icon_state = "generic"
+	icon = 'icons/obj/machines/vending/generic.dmi'
+	icon_state = ICON_STATE_WORLD
```

Comment:
I really don't get the point of using `ICON_STATE_WORLD` for non-objects. An inventory or worn sprite is never valid here. It's not like vendors are the only things that do this, obviously, but still, I don't rrrrrrrrreally get it?